apis/Google.Cloud.Firestore.Data/Google.Cloud.Firestore.Data.Tests/Proto/ProtoTest.cs

```diff
@@ -0,0 +1,255 @@
+        // In theory, this should allow all tests to be shown in VS too, but that doesn't appear to work right now.
+        public class SerializableTest : IXunitSerializable
+        {
+            public Test Test { get; private set; }
+
+            // Used in deserialization
+            public SerializableTest() { }
+
+            public SerializableTest(Test test) => Test = test;
+
+            public void Deserialize(IXunitSerializationInfo info) =>
+                Test = Test.Parser.ParseFrom(info.GetValue<byte[]>(nameof(Test)));
+
+            public void Serialize(IXunitSerializationInfo info) =>
+                info.AddValue(nameof(Test), Test.ToByteArray());
+
+            public override string ToString() => Test?.Description ?? "(Unknown)";
+        }        
+
+        private static object DeserializeJson(string json)
+        {
+            wkt::Value proto = wkt::Value.Parser.ParseJson(json);
+            return ConvertValue(proto);
+        }
+
+        /// <summary>
+        /// Converts a value from well-known type format to .NET CLR format (using dictionaries and lists).
+        /// </summary>
+        private static object ConvertValue(wkt::Value value)
+        {
+            switch (value.KindCase)
+            {
+                case wkt::Value.KindOneofCase.BoolValue:
+                    return value.BoolValue;
+                case wkt::Value.KindOneofCase.ListValue:
+                    return value.ListValue.Values.Select(ConvertValue).ToArray();
+                case wkt::Value.KindOneofCase.NullValue:
+                    return null;
+                case wkt::Value.KindOneofCase.NumberValue:
+                    // Unfortunately we lose the "integer vs floating point" detail from the JSON.
+                    // Let's assume that anything which rounds to itself is an integer.
+                    var number = value.NumberValue;
+                    return Math.Round(number) == number ? (object) (long) number : (object) number;
+                case wkt::Value.KindOneofCase.StringValue:
+                    return value.StringValue == "ServerTimestamp" ? (object) SentinelValue.ServerTimestamp
+                        : value.StringValue == "Delete" ? (object) SentinelValue.Delete
+                        : (object) value.StringValue;
+                case wkt::Value.KindOneofCase.StructValue:
+                    return value.StructValue.Fields.ToDictionary(pair => pair.Key, pair => ConvertValue(pair.Value));
+                default:
+                    throw new ArgumentException($"Unknown value kind: {value.KindCase}");
+            }
+        }
+
+        // Rather than using Moq to expect a commit, we just have a small recording client. This makes
+        // it easier to get good diagnostics about what went wrong.
+        private class MockCommitClient : FirestoreClient
+        {
+            private readonly CommitRequest m_expectedRequest;
```

Comment:
Why the "m" prefix?

Reply:
Whoops, had forgotten our convention. (I've used too many conventions in too many different places...)